OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java

```diff
@@ -769,7 +771,22 @@ public synchronized NextDirectionInfo getNextRouteDirectionInfo(NextDirectionInf
 	}
 
 	public synchronized float getCurrentMaxSpeed() {
```

Comment:
nope this is not correct fix, cause maxspeed should be properly calculated inside route and not in Routing Helper.

So it should maxspeed should be propagated from RouteCalculateResult somehow and influenced by RouteResultPreparation.